state/charm.go

```diff
@@ -52,14 +52,7 @@ type charmDoc struct {
 
 	// Life manages charm lifetime in the usual way, but only local
 	// charms can actually be "destroyed"; store charms are
-	// immortal. When a local charm is removed, its document is left
-	// in place, with Life set to Dead, to ensure we don't
-	// accidentally reuse the charm URL, which must be unique within
-	// a model.
-	//
-	// Note that this aligns with the existing contract implied by
-	// Dead: that most clients should see it as not existing at all.
-	// Nothing strictly obliges us to clean up the doc.
+	// immortal.
```

Comment:
There should be changes to the charm unit tests to ensure that removing the dead docs still sees the correct revision being used.